<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>nf-core/atacseq &mdash; BovReg workshop ATAC-seq data analaysis  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="nf-core" href="nf-core.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> BovReg workshop ATAC-seq data analaysis
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About the course</a></li>
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="nf-core.html">nf-core</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">nf-core/atacseq</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pipeline-summary">Pipeline summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run-nf-core-atacseq-with-test-data">Run nf-core/atacseq with test data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#samplesheet-input">Samplesheet input</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#multiple-runs-of-the-same-sample">Multiple runs of the same sample</a></li>
<li class="toctree-l3"><a class="reference internal" href="#full-samplesheet">Full samplesheet</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#parameters">Parameters</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">BovReg workshop ATAC-seq data analaysis</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a></li>
      <li class="breadcrumb-item active">nf-core/atacseq</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="nf-core-atacseq">
<span id="atac-seq-pipeline-page"></span><h1>nf-core/atacseq<a class="headerlink" href="#nf-core-atacseq" title="Permalink to this heading"></a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this heading"></a></h2>
<p><a class="reference external" href="https://nf-co.re/atacseq">nfcore/atacseq</a> is a bioinformatics best-practice pipeline used for the analysis of ATAC-seq data. As discussed before,
the pipeline is built using <a class="reference external" href="https://www.nextflow.io/">Nextflow</a>, a workflow manager to run tasks across multiple compute infrastructures in a
very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible.
The <a class="reference external" href="https://www.nextflow.io/docs/latest/dsl2.html">Nextflow DSL2</a> implementation of this pipeline uses one container per process which makes
it much easier to maintain and update software dependencies. Where possible, these processes have been submitted to and installed from
<a class="reference external" href="https://github.com/nf-core/modules">nf-core/modules</a> in order to make them available to all nf-core pipelines, and
to everyone within the Nextflow community.</p>
</section>
<section id="pipeline-summary">
<h2>Pipeline summary<a class="headerlink" href="#pipeline-summary" title="Permalink to this heading"></a></h2>
<a class="reference internal image-reference" href="_images/metro_map_atacseq.png"><img alt="_images/metro_map_atacseq.png" src="_images/metro_map_atacseq.png" style="width: 600px;" /></a>
<p>The main functionality of the nf-core/atacseq is summarised in the figure above and listed in the following numbered list:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Raw read QC (FastQC)</p></li>
<li><p>Adapter trimming (Trim Galore!)</p></li>
<li><dl class="simple">
<dt>Choice of multiple aligners</dt><dd><p>1.(BWA)
2.(Chromap) For paired-end reads only working until mapping step since the <a class="reference external" href="https://github.com/nf-core/chipseq/issues/291">resulting BAMs are faulty</a>
3.(Bowtie2)
4.(STAR)</p>
</dd>
</dl>
</li>
<li><p>Mark duplicates (picard)</p></li>
<li><dl class="simple">
<dt>Merge alignments from multiple libraries of the same sample (picard)</dt><dd><ol class="arabic simple">
<li><p>Re-mark duplicates (picard)</p></li>
<li><dl class="simple">
<dt>Filtering to remove:</dt><dd><ul class="simple">
<li><p>reads mapping to mitochondrial DNA (SAMtools)</p></li>
<li><p>reads mapping to blacklisted regions (SAMtools, BEDTools)</p></li>
<li><p>reads that are marked as duplicates (SAMtools)</p></li>
<li><p>reads that arent marked as primary alignments (SAMtools)</p></li>
<li><p>reads that are unmapped (SAMtools)</p></li>
<li><p>reads that map to multiple locations (SAMtools)</p></li>
<li><p>reads containing &gt; 4 mismatches (BAMTools)</p></li>
<li><p>reads that are soft-clipped (BAMTools)</p></li>
<li><p>reads that have an insert size &gt; 2kb (BAMTools; paired-end only)</p></li>
<li><p>reads that map to different chromosomes (Pysam; paired-end only)</p></li>
<li><p>reads that arent in FR orientation (Pysam; paired-end only)</p></li>
<li><p>reads where only one read of the pair fails the above criteria (Pysam; paired-end only)</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Alignment-level QC and estimation of library complexity (picard, Preseq)</p></li>
<li><p>Create normalised bigWig files scaled to 1 million mapped reads (BEDTools, bedGraphToBigWig)</p></li>
<li><p>Generate gene-body meta-profile from bigWig files (deepTools)</p></li>
<li><p>Calculate genome-wide enrichment (deepTools)</p></li>
<li><p>Call broad/narrow peaks (MACS2)</p></li>
<li><p>Annotate peaks relative to gene features (HOMER)</p></li>
<li><p>Create consensus peakset across all samples and create tabular file to aid in the filtering of the data (BEDTools)</p></li>
<li><p>Count reads in consensus peaks (featureCounts)</p></li>
<li><p>Differential accessibility analysis, PCA and clustering (R, DESeq2)</p></li>
<li><p>Generate ATAC-seq specific QC html report (ataqv)</p></li>
</ol>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Merge filtered alignments across replicates (picard)</dt><dd><ol class="arabic simple">
<li><p>Re-mark duplicates (picard)</p></li>
<li><p>Remove duplicate reads (SAMtools)</p></li>
<li><p>Create normalised bigWig files scaled to 1 million mapped reads (BEDTools, bedGraphToBigWig)</p></li>
<li><p>Call broad/narrow peaks (MACS2)</p></li>
<li><p>Annotate peaks relative to gene features (HOMER)</p></li>
<li><p>Create consensus peakset across all samples and create tabular file to aid in the filtering of the data (BEDTools)</p></li>
<li><p>Count reads in consensus peaks relative to merged library-level alignments (featureCounts)</p></li>
<li><p>Differential accessibility analysis, PCA and clustering (R, DESeq2)</p></li>
</ol>
</dd>
</dl>
</li>
<li><p>Create IGV session file containing bigWig tracks, peaks and differential sites for data visualisation (IGV)</p></li>
<li><p>Present QC for raw read, alignment, peak-calling and differential accessibility results (ataqv, MultiQC, R)</p></li>
</ol>
</div></blockquote>
</section>
<section id="run-nf-core-atacseq-with-test-data">
<h2>Run nf-core/atacseq with test data<a class="headerlink" href="#run-nf-core-atacseq-with-test-data" title="Permalink to this heading"></a></h2>
<p>As discussed previously <a class="reference external" href="https://bovreg.github.io/atacseq-workshop-limoges/nf-core.html#running-pipelines-with-test-data">here</a>
nf-core pipelines include a special configuration named <code class="docutils literal notranslate"><span class="pre">test</span></code> that enables to run the whole pipeline with a small subsampled
dataset. Since the AWS environment provided has not enough computational resources, we will use this test profile to showcase
the pipeline functionality during this tutorial.</p>
<p>We will first launch the pipeline using the nf-core <code class="docutils literal notranslate"><span class="pre">launch</span></code> command and the minimal set of parameters to run the nf-core/atacseq
pipeline with the test data set. Below, you will find several snippets to guide to chose the correct parametrization during this process:</p>
<ul>
<li><p>Call the nf-core tools launch command:</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">nf-core launch</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Make sure you chose the development version of the pipeline:</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">Select release / branch: (Use arrow keys)</span>
<span class="go">    1.2.2  [release]</span>
<span class="go">    1.2.1  [release]</span>
<span class="go">    1.2.0  [release]</span>
<span class="go">    1.1.0  [release]</span>
<span class="go">    1.0.0  [release]</span>
<span class="go">    » dev  [branch]</span>
<span class="go">    master  [branch]</span>
<span class="go">    multiqc_fail  [branch]</span>
<span class="go">    schema  [branch]</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Then, make sure that you set the <code class="docutils literal notranslate"><span class="pre">test</span></code> and the <code class="docutils literal notranslate"><span class="pre">docker</span></code> profiles. The former will load the settings for the test dataset run and the latter will set the settings for launching the pipeline using <a class="reference external" href="https://www.docker.com/">Docker</a>.</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">?  Nextflow command-line flags</span>
<span class="go">General Nextflow flags to control how the pipeline runs.</span>
<span class="go">These are not specific to the pipeline and will not be saved in any parameter file. They are just used when building the nextflow run launch command.</span>
<span class="gp gp-VirtualEnv">(Use arrow keys)</span>

<span class="go">» Continue &gt;&gt;</span>
<span class="go">---------------</span>
<span class="go">-name</span>
<span class="go">-profile  [test,docker]</span>
<span class="go">-work-dir  [./work]</span>
<span class="go">-resume  [False]</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Now, provide the <code class="docutils literal notranslate"><span class="pre">required</span></code> parameters, the <code class="docutils literal notranslate"><span class="pre">input</span></code> for the input samplesheet and the <code class="docutils literal notranslate"><span class="pre">outdir</span></code> to dump the pipeline results.</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">?  Input/output options</span>
<span class="go">Define where the pipeline should find input data and save output data.</span>
<span class="gp gp-VirtualEnv">(Use arrow keys)</span>
<span class="go"> » Continue &gt;&gt;</span>
<span class="go">    ---------------</span>
<span class="go">    input  [https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/samplesheet/v2.0/samplesheet_test.csv]</span>
<span class="go">    fragment_size  [200]</span>
<span class="go">    seq_center</span>
<span class="go">    read_length</span>
<span class="go">    outdir  [results_test]</span>
<span class="go">    email</span>
<span class="go">    multiqc_title</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Finally, we will just left all the rest of parameters set as default until reaching the final prompt:</p>
<blockquote>
<div><div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp gp-VirtualEnv">(Use arrow keys)</span>
<span class="go">  Continue &gt;&gt;</span>
<span class="go">INFO     [✓] Input parameters look valid                                                                                                          schema.py:213</span>
<span class="go">INFO     Nextflow command:                                                                                                                        launch.py:724</span>
<span class="go">        nextflow run nf-core/atacseq -r dev -profile &quot;test,docker&quot; -params-file &quot;nf-params.json&quot;</span>


<span class="go">Do you want to run this command now?  [y/n] (y):</span>
</pre></div>
</div>
</div></blockquote>
</li>
</ul>
</section>
<section id="samplesheet-input">
<h2>Samplesheet input<a class="headerlink" href="#samplesheet-input" title="Permalink to this heading"></a></h2>
<p>To provide the samples to the pipeline, you will need to create a samplesheet with with information about
the samples you would like to analyse as the ones shown below.</p>
<p>The file above, is a comma-separated values (CSV) file with three columns. It can be provided to the pipeline using
the <code class="docutils literal notranslate"><span class="pre">input</span></code> parameter.</p>
<section id="multiple-runs-of-the-same-sample">
<h3>Multiple runs of the same sample<a class="headerlink" href="#multiple-runs-of-the-same-sample" title="Permalink to this heading"></a></h3>
<p>If you have re-sequenced the same sample several times to increase its sequencing depth, you can input this information
to the pipeline by using the same sample identifier. The pipeline will concatenate the raw reads before performing any
downstream analysis. Below is an example for the same sample sequenced across 3 lanes:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sample,fastq_1,fastq_2</span>
<span class="go">CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz</span>
<span class="go">CONTROL_REP1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz</span>
<span class="go">CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz</span>
</pre></div>
</div>
</section>
<section id="full-samplesheet">
<h3>Full samplesheet<a class="headerlink" href="#full-samplesheet" title="Permalink to this heading"></a></h3>
<p>The samplesheet also can encode the information or whether a sample is single- or paired-end. Single-end samples
can be provided by just letting empty the field corresponding to the <code class="docutils literal notranslate"><span class="pre">fastq_2</span></code> column.
The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 3 columns to
match those defined in the table below.</p>
<p>A final samplesheet file consisting of both single- and paired-end data may look something like the one below.
This is for 6 samples, where TREATMENT_REP3 has been sequenced twice:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">sample,fastq_1,fastq_2</span>
<span class="go">CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz</span>
<span class="go">CONTROL_REP2,AEG588A2_S2_L002_R1_001.fastq.gz,AEG588A2_S2_L002_R2_001.fastq.gz</span>
<span class="go">CONTROL_REP3,AEG588A3_S3_L002_R1_001.fastq.gz,AEG588A3_S3_L002_R2_001.fastq.gz</span>
<span class="go">TREATMENT_REP1,AEG588A4_S4_L003_R1_001.fastq.gz,</span>
<span class="go">TREATMENT_REP2,AEG588A5_S5_L003_R1_001.fastq.gz,</span>
<span class="go">TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,</span>
<span class="go">TREATMENT_REP3,AEG588A6_S6_L004_R1_001.fastq.gz,</span>
</pre></div>
</div>
</section>
</section>
<section id="parameters">
<h2>Parameters<a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p>You can find the full list of parameters <a class="reference external" href="https://nf-co.re/atacseq/dev/parameters">here</a></p>
<p>different types of parameters</p>
<p>skip parameters etc</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="nf-core.html" class="btn btn-neutral float-left" title="nf-core" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, CRG.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>